exonum/src/api/node/public/explorer.rs

```diff
@@ -276,6 +351,44 @@ impl ExplorerApi {
             })
     }
 
+    /// Return call status of committed transaction.
+    pub fn call_status(
+        snapshot: &dyn Snapshot,
+        query: CallStatusQuery,
+    ) -> Result<CallStatusResponse, ApiError> {
+        let explorer = BlockchainExplorer::new(snapshot);
+
+        let (call_in_block, height) = match query {
+            CallStatusQuery::CallWithHeight { height, call } => (call.into(), Height(height)),
+            CallStatusQuery::TransactionHash { tx_hash } => {
+                let tx_info = explorer.transaction(&tx_hash).ok_or_else(|| {
+                    ApiError::NotFound(format!("Unknown transaction hash ({})", tx_hash))
+                })?;
+                let tx_info = match tx_info {
+                    TransactionInfo::Committed(info) => Ok(info),
+                    TransactionInfo::InPool { .. } => Err(ApiError::NotFound(format!(
+                        "Requested transaction ({}) is not executed yet",
+                        tx_hash
+                    ))),
+                }?;
```

Comment:
Nit: Calling `?` here looks weird, IMO. Something like this could be more idiomatic:

```rust
let tx_info = match tx_info {
    TransactionInfo::Committed(info) => info,
    TransactionInfo::InPool { .. } => {
        let err = ApiError::NotFound(/* ... */);
        return Err(err);
    }
};
```